### ISSUE TITLE
Integration with Solidus

### ISSUE THREAD

**Issue by dinohamzic** [vercel/commerce#72]
At Nebulab we are working on creating a custom open source storefront for [Solidus](https://solidus.io) based on Next.js and GraphQL.

After watching Next.js Conf and seeing Next.js Commerce in action, we started an internal discussion to understand if it would be feasible (and an overall sensible move) to try to integrate Solidus with this storefront instead of making one from scratch. 

As per our understanding, you are planning to have out of the box integrations with different eCommerce backends. When adding a new integration, we imagine the ideal workflow to be to simply install the appropriate backend integration in this project, like the folks at BigCommerce did [there](https://github.com/bigcommerce/storefront-data-hooks), and to leave the code here untouched. In our minds, this approach would delegate the maintenance of the storefront UI to you, and allow us to solely focus on the backend integration logic.

However, after playing with this beautiful demo and having a look at the code, we are unsure on how to proceed. We would like to know your thoughts on some of our doubts and questions:
- in general, how do you envision a different backend to be integrated in the current storefront?
- `bigcommerce/storefront-data-hooks` is currently hardcoded throughout the codebase, how would integrating a different `storefront-data-hooks` look like? maybe an installer?
- what about unsupported features (for example, Solidus does not support a wishlist out of the box)? how would you handle this case? 

To sum up, we would like to have a high level overview and discusson on how to approach the integration problem.

Thanks,
The Nebulab Team

**Comment by DeanDeBlock** [comment]
Hi, @dinohamzic, I am currently working out a proof of concept for the Solidus (Graphql extension) backend + Commerce storefront

The commerce codebase relies on the shape of the Graphql's responses (BigCommerce), so I will probably be making some kind of translation for the Solidus graphql responses

Are you also working on this?

**Comment by okbel** [comment]
@DeanDeBlock Could you please check the agnostic branch?

**Comment by DeanDeBlock** [comment]
Looking good @okbel !

let me try and get a test version up and running and I'll get back to you

Cheers

**Comment by DeanDeBlock** [comment]
Hi @okbel ok Still some things to do, but getting there https://solidus-vercel-front.herokuapp.com/

How do we go from here, shall i request a pull on the `canary` branch from my fork ?

**Comment by manovotny** [comment]
Hey there! Thank you for your contribution. We have decided to take Next.js Commerce in a new direction and will be closing out current PRs and issues due to this change. Please see this PR for more details: #966
